oak-run/src/main/java/org/apache/jackrabbit/oak/run/CreateGarbageCommand.java

```diff
@@ -0,0 +1,574 @@
+
+        try (Closer closer = Closer.create()) {
+            // Register the ExecutorService with Closer
+            closer.register(() -> {
+                // Properly shutdown the executor service
+                continuousRunExecutor.shutdown();
+                try {
+                    if (!continuousRunExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
+                        continuousRunExecutor.shutdownNow();
+                    }
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            execute(closer, args);
+
+        } catch (Throwable e) {
+            LOG.error("Command failed", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Method with passed closer is necessary in order to allow for unit tests to check the output of the command.
+     * It is the responsibility of the caller to close the closer.
+     *
+     * Returns the list of generated garbage base paths (under the garbage root node).
+     * @param closer
+     * @param args
+     * @throws Exception
+     */
+    public List<String> execute(Closer closer, String... args) throws Exception {
+        continuousRunIndex = 0;
+
+        List<String> generateBasePaths = new ArrayList<>();
+
+        CreateGarbageOptions options = new CreateGarbageOptions(USAGE).parse(args);
+        String subCmd = options.getSubCmd();
+
+        LOG.info("Executing command: " + getClass().getName() + " with sub-command: " + subCmd);
+
+        if (CreateGarbageOptions.CMD_GENERATE.equals(subCmd)) {
+
+            if (GenerateGarbageHelper.isInvalidGarbageGenerationMode(options.getGarbageType())) {
+                LOG.error("Invalid fullGCMode specified: " + options.getGarbageType() + " in: " + getClass().getName());
+                System.exit(1);
+            }
+            if (options.getCreateGarbageNodesCount() <= 0) {
+                LOG.error("Invalid garbageNodesCount specified: " + options.getCreateGarbageNodesCount() + " in: " + getClass().getName());
+                System.exit(1);
+            }
+            if (options.getGarbageNodesParentCount() <= 0) {
+                LOG.error("Invalid garbageNodesParentCount specified: " + options.getGarbageNodesParentCount() + " in: " + getClass().getName());
+                System.exit(1);
+            }
+
+            if (options.getNumberOfRuns() > 1 && options.getGenerateIntervalSeconds() > 0) {
+                generateBasePaths.addAll(generateGarbageContinuously(options, closer));
+            } else {
+                generateBasePaths.add(createGarbage(options, closer, 0));
+            }
+        } else if (CreateGarbageOptions.CMD_CLEAN.equals(subCmd)) {
+            cleanGarbage(options, closer);
+        } else {
+            System.err.println("Unknown revisions command: " + subCmd);
+        }
+
+        return generateBasePaths;
+    }
+
+    @Override
+    public void close() throws IOException {
```

Comment:
We can safely remove this since we are registering `continuousRunExecutor` with a `closer` instance.

Reply:
Done, thank you for the suggestion.